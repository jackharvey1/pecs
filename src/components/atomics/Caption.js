import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const TextArea = styled.textarea`
    width: 125px;

    @media (min-width: 1024px) {
        width: 150px;
    }

    margin-top: 12px;

    border: none;
    color: #333333;

    font-size: 16px;
    font-family: Helvetica, Arial, sans-serif;
    text-align: center;

    resize: none;

    &:focus {
        outline: none;
    }

    @media print {
        &::placeholder {
            color: transparent;
        }
    }
`;

const Caption = ({ addToCaption, caption, index }) => (
    <TextArea
        id="caption"
        rows="3"
        cols="10"
        value={caption}
        onChange={(event) => addToCaption(index, event)}
        placeholder="Add caption here"
    />);

Caption.propTypes = {
    index: PropTypes.number.isRequired,
    caption: PropTypes.string,
    addToCaption: PropTypes.func.isRequired,
};

export default Caption;

