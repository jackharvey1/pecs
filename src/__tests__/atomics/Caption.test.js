import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Caption from '../../components/atomics/Caption';

describe('Caption component', () => {
    it('renders as intended', () => {
        const wrapper = renderer.create(<Caption
            index={0}
            caption="Some text"
            addToCaption={() => null}
        />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('sets the text as the passed prop', () => {
        const wrapper = shallow(<Caption
            index={0}
            caption="Expected text"
            addToCaption={() => null}
        />);
        expect(wrapper.props().value).toEqual('Expected text');
    });

    it('invokes on change with the proper parameters', () => {
        const mockOnChange = jest.fn();
        const event = { target: { value: 'caption' } };
        const wrapper = shallow(<Caption
            index={1}
            caption="captio"
            addToCaption={mockOnChange}
        />);
        wrapper.simulate('change', event);
        expect(mockOnChange).toHaveBeenCalledWith(1, event);
    });
});
