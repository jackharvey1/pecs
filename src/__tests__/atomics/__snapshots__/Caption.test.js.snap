// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption component renders as intended 1`] = `
.c0 {
  width: 125px;
  margin-top: 12px;
  border: none;
  color: #333333;
  font-size: 16px;
  font-family: Helvetica,Arial,sans-serif;
  text-align: center;
  resize: none;
}

.c0:focus {
  outline: none;
}

@media (min-width:1024px) {
  .c0 {
    width: 150px;
  }
}

@media print {
  .c0::-webkit-input-placeholder {
    color: transparent;
  }

  .c0::-moz-placeholder {
    color: transparent;
  }

  .c0:-ms-input-placeholder {
    color: transparent;
  }

  .c0::placeholder {
    color: transparent;
  }
}

<textarea
  className="c0"
  cols="10"
  id="caption"
  onChange={[Function]}
  placeholder="Add caption here"
  rows="3"
  value="Some text"
/>
`;
